test/rhsmlib_test/test_file_monitor.py

```diff
@@ -0,0 +1,242 @@
+from __future__ import print_function, division, absolute_import
+
+from rhsmlib import file_monitor
+from six.moves import configparser
+from mock import Mock, patch
+from test import fixture
+import subprocess
+from threading import Thread
+
+
+class TestFilesystemWatcher(fixture.SubManFixture):
+    def setUp(self):
+        super(TestFilesystemWatcher, self).setUp()
+        self.mock_cb1 = Mock(return_value=None)
+        self.mock_cb2 = Mock(return_value=None)
+        self.testpath1 = self.write_tempfile("").name
+        self.testpath2 = self.write_tempfile("").name
+        self.testpath3 = self.write_tempfile("").name
+        self.dw1 = file_monitor.DirectoryWatch(self.testpath1, [], glob=True)
+        self.dw2 = file_monitor.DirectoryWatch(self.testpath2, [], glob=False)
+        self.dw3 = file_monitor.DirectoryWatch(self.testpath2, [self.mock_cb1, self.mock_cb2], glob=False)
+        self.dir_list = [self.dw1, self.dw2, self.dw3]
+        self.fsw1 = file_monitor.FilesystemWatcher(self.dir_list)
+        self.fsw2 = file_monitor.InotifyFilesystemWatcher(self.dir_list)
+
+    def tearDown(self):
+        super(TestFilesystemWatcher, self).tearDown()
+
+    @patch("rhsmlib.file_monitor.is_inotify_available")
+    @patch("rhsmlib.file_monitor.is_inotify_config")
+    def test_get_fsw_version(self, mock_config, mock_avail):
```

Comment:
Add doc string to test method or split test method into more methods with self-descriptive name.